pkg/geoipupdate/geoip_updater.go

```diff
@@ -4,22 +4,61 @@ package geoipupdate
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
-	"log"
+	"os"
+	"sync"
+	"time"
 
 	"github.com/maxmind/geoipupdate/v5/pkg/geoipupdate/database"
 	"github.com/maxmind/geoipupdate/v5/pkg/geoipupdate/internal"
+	"github.com/maxmind/geoipupdate/v5/pkg/geoipupdate/vars"
+)
+
+var (
+	output = vars.NewBareDiscardLogger()
+	log    = vars.NewDiscardLogger("updater")
 )
 
 // Client uses config data to initiate a download or update
 // process for GeoIP databases.
 type Client struct {
-	config *Config
+	config    *Config
+	getReader func() (database.Reader, error)
+	getWriter func() (database.Writer, error)
 }
 
 // NewClient initialized a new Client struct.
 func NewClient(config *Config) *Client {
-	return &Client{config: config}
+	if config.Verbose {
+		output.SetOutput(os.Stdout)
+		log.SetOutput(os.Stderr)
+	}
```

Comment:
These sorts of mutations of the package loggers when calling these functions seems unexpected. I think ideally the loggers wouldn't be package variables.  For `NewClient` in particular, maybe we could allow passing the logger in as part of `*Config` so that users of the library could use whatever logger is appropriate for their use case.